genie-core/src/main/java/com/netflix/genie/core/jpa/services/JpaFileServiceImpl.java

```diff
@@ -74,7 +75,13 @@ public void createFileIfNotExists(@NotBlank(message = "File path cannot be blank
      * {@inheritDoc}
      */
     @Override
-    public int deleteUnusedFiles(@NotNull final Instant createdThreshold) {
-        return this.fileRepository.deleteUnusedFiles(Date.from(createdThreshold));
+    public long deleteUnusedFiles(@NotNull final Instant createdThreshold) {
+        return this.fileRepository.deleteByIdIn(
```

Comment:
Do you want to batch it or does JPA handle that?

Reply:
given the scale of this data I was trying to avoid batching it